src/main/scala/io/qbeast/spark/delta/OTreeIndex.scala

```diff
@@ -26,19 +29,40 @@ case class OTreeIndex(index: TahoeLogFileIndex)
 
   private def qbeastSnapshot = DeltaQbeastSnapshot(snapshot)
 
-  override def matchingFiles(
-      partitionFilters: Seq[Expression],
-      dataFilters: Seq[Expression]): Seq[AddFile] = {
+  protected def absolutePath(child: String): Path = {
+    val p = new Path(new URI(child))
+    if (p.isAbsolute) {
+      p
+    } else {
+      new Path(index.path, p)
+    }
+  }
 
-    val tahoeMatchingFiles = index.matchingFiles(partitionFilters, dataFilters)
-    val previouslyMatchedFiles =
-      tahoeMatchingFiles.map(addFile => QbeastFile(addFile.path, addFile.tags))
+  def matchingFiles(
```

Comment:
this method should be private, and probably better change its name, so it can't be confused with the TahoeIndex's method.